corehq/apps/api/query_adapters.py

```diff
@@ -53,7 +53,7 @@ def count(self):
     def __getitem__(self, item):
         if isinstance(item, slice):
             limit = item.stop - item.start
-            result = GroupES().domain(self.domain).sort('name.exact').size(limit).start(item.start).run()
+            result = GroupES().domain(self.domain).sort('_id').size(limit).start(item.start).run()
```

Comment:
This will make it sort by ID instead of by name, right? Is there a way to keep this sorted by name? This is for an external API and I imagine some people might rely on that. In any case, if we change the order here we may have to consider announcing the change somehow, which wouldn't be the end of the world.

Reply:
Yes, that's right. It's possible to sort the final result sorted by name in python, but not retrieve slices from ES in sort-by-name order. The main reason this is sorted it to support pagination and sorting by name is not really a documented feature https://confluence.dimagi.com/display/commcarepublic/List+Groups. The number of requests on prod for this API was around [12 in last six months](https://analytics.google.com/analytics/web/#/report/content-pages/a22103010w43589809p43613522/_u.date00=20200101&_u.date01=20200802&explorer-table.filter=.*~2Fapi~2F.*group.*&explorer-table.plotKeys=%5B%5D/). So, I feel  an announcement is not super necessary. But if we have to, I guess commcare-users Google group?

Reply:
I think google analytics might only be reported from the browser, so it wouldn't capture API requests well. I checked our access logs and it's under very heavy usage, i.e. more than a thousand requests in the last 12h hours.

Ideally we'd find a way to sort by name just to keep things from changing, but if not the most important thing is to make sure the sort order is consistent. (If the sort order isn't consistent then you can't meaningfully paginate through results.)

Reply:
I added a few more commits here that (1) add a test for sort order and (2) switch back to using `name.exact` here, which appears to work despite the mapping changes in this PR. Could you review them to see if I'm missing anything?

The only change now is that it went from a case-insensitive sort (a, b, c, d, E) to a case-sensitive sort (E, a, b, c, d). I also added a test followed by a test change to reflect this.